src/components/ResortsData.js

```diff
@@ -18,30 +20,37 @@ class ResortsData extends Component{
     this.fetchResorts();
   }
 
-  componentWillUpdate(nextProps, nextState) {
-    if (nextState.page !== this.state.page) {
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.page !== this.state.page) {
       this.fetchResorts();
     }
   }
 
   fetchResorts(){
     this.setState({isLoading: true})
-    fetch(`http://localhost:3001/resorts?_page=${this.state.page}&_limit=1`)
+    fetch(`http://localhost:3001/resorts?_page=${this.state.page}&_limit=${this.state.perPage}`)
         .then(response => {
           if (response.ok) {
-            return response.json();
+            return response.json().then((data) => ({data, headers: response.headers}))
           } else {
             throw new Error('Something went wrong ...');
           }
         })
-        .then(data => this.setState({ data: data, isLoading: false }))
+        .then(({data, headers}) => {
+          this.setState({ data: data,  pageCount: Math.ceil(headers.get('x-total-count')/ this.state.perPage), isLoading: false })
+          console.log(`page: ${this.state.page}`)
+        })
         .catch(error => this.setState({ error, isLoading: false}))
   }
 
-  handleClick = (page) => {this.setState({page},() => this.fetchResorts());}
+  handlePageClick = (data) => {
+    console.log(`selected : ${data.selected}`)
+    //there is +1, because 'react-paginate' has 0 as start page number,
+    this.setState({page: data.selected +1})
+  };
 
   render(){
-    const {isLoading, data, error} = this.state;
+    const {data, pageCount, isLoading, error} = this.state;
     if(error)
       return(error.message)
     if(isLoading)
```

Comment:
When you click pagination it triggers loading flag, and the whole main return (which contains pagination) is removed and replaced with `Loading in progress`.
When data is loaded, main return (and Pagination component) is rendered again, and by default the page number `1` is selected. That's why it was always displayed regardless of what you clicked.

BTW If you click twice on the same page, the active button properly changes.

Reply:
thx!
yes I knew that when I click twice 'active' is triggered :)